lib/model.js

```diff
@@ -1454,10 +1454,17 @@ class Model {
     options = Utils.cloneDeep(options);
 
     if (options.limit === undefined) {
-      const pkVal = options.where && options.where[this.primaryKeyAttribute];
 
-      // Don't add limit if querying directly on the pk
-      if (!options.where || !(Utils.isPrimitive(pkVal) || Buffer.isBuffer(pkVal))) {
+      var self = this;
+      // Don't add limit if querying directly on the pk or any other unique key
+      if (!options.where || !Object.keys(this.attributes).some(function (key) {
+        var attribute = self.attributes[key];
+        if (attribute.unique || attribute.primaryKey) {
```

Comment:
`unique` can be a composite unique, so you'll need to check that it's a boolean value